packages/framework/src/client.test.ts

```diff
@@ -51,6 +51,14 @@ describe('Novu Client', () => {
       process.env = { ...process.env, NODE_ENV: originalEnv };
     });
 
+    it('should set strictAuthentication to false when NODE_ENV is not defined', () => {
+      const originalEnv = process.env.NODE_ENV;
+      process.env = { ...process.env, NODE_ENV: undefined as any };
+      const newClient = new Client({ secretKey: 'some-secret-key' });
+      expect(newClient.strictAuthentication).toBe(false);
+      process.env = { ...process.env, NODE_ENV: originalEnv };
+    });
+
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review. Do not leave false praise (which can actually be damaging). Do look for something to sincerely praise.  -->
👏 **praise:** Excellent!‏